src/components/Button/BaseButton.tsx

```diff
@@ -57,6 +59,12 @@ export const BaseButton: FC<InternalButtonProps> = React.forwardRef(
         const smallScreenActive: boolean = useMatchMedia(Breakpoints.Small);
         const xSmallScreenActive: boolean = useMatchMedia(Breakpoints.XSmall);
 
+        const contextuallyDisabled: DisabledType = useContext(DisabledContext);
+        const mergedDisabled: boolean = contextuallyDisabled || disabled;
+
+        const contextuallySized: SizeType = useContext(SizeContext);
+        const mergedSize = contextuallySized || size;
```

Comment:
NoFormStyle only is relevant to validation styles. What I can do is add another shared set of `boolean` prop to components that may be used in a form as a mechanism to avoid contextual size and shape. Perhaps `noDisabledContext`  `noSizeContext` and `noShapeContext` in an enum called `contextProps` via the config provider types?

Reply:
added.